mmdet/models/dense_heads/centernet_head.py

```diff
@@ -0,0 +1,378 @@
+from mmcv.runner import force_fp32
+
+from mmdet.core import multi_apply
+from mmdet.models import HEADS, build_loss
+from mmdet.models.utils import gaussian_radius, gen_gaussian_target
+from ..utils.gaussian_target import (get_local_maximum, get_topk_from_heatmap,
+                                     transpose_and_gather_feat)
+from .base_dense_head import BaseDenseHead
+
+
+@HEADS.register_module()
+class CenterNetHead(BaseDenseHead):
+    """Objects as Points Head. CenterHead use center_point to indicate object's
+    position. Paper link <https://arxiv.org/abs/1904.07850>
+
+    Args:
+        in_channel (int): Number of channel in the input feature map.
+        feat_channel (int): Number of channel in the intermediate feature map.
+        num_classes (int): Number of categories excluding the background
+            category.
+        loss_center_heatmap (dict | None): Config of center heatmap loss.
+            Default: GaussianFocalLoss.
+        loss_wh (dict | None): Config of wh loss. Default: L1Loss.
+        loss_offset (dict | None): Config of offset loss. Default: L1Loss.
+        train_cfg (dict | None): Training config. Useless in CenterNet,
+            but we keep this variable for SingleStageDetector. Default: None.
+        test_cfg (dict | None): Testing config of CenterNet. Default: None.
+        init_cfg (dict or list[dict], optional): Initialization config dict.
+            Default: None
+    """
+
+    def __init__(self,
+                 in_channel,
+                 feat_channel,
+                 num_classes,
+                 loss_center_heatmap=dict(
+                     type='GaussianFocalLoss', loss_weight=1.0),
+                 loss_wh=dict(type='L1Loss', loss_weight=0.1),
+                 loss_offset=dict(type='L1Loss', loss_weight=1.0),
+                 train_cfg=None,
+                 test_cfg=None,
+                 init_cfg=None):
+        super(CenterNetHead, self).__init__(init_cfg)
+        self.num_classes = num_classes
+        self.heatmap_head = self._build_head(
+            in_channel,
+            feat_channel,
+            num_classes,
+        )
+        self.wh_head = self._build_head(in_channel, feat_channel, 2)
+        self.offset_head = self._build_head(in_channel, feat_channel, 2)
+
+        self.loss_center_heatmap = build_loss(loss_center_heatmap)
+        self.loss_wh = build_loss(loss_wh)
+        self.loss_offset = build_loss(loss_offset)
+
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+
+    def _build_head(self, in_channel, feat_channel, out_channel):
+        """Build head for each branch."""
+        layer = nn.Sequential(
+            nn.Conv2d(in_channel, feat_channel, kernel_size=3, padding=1),
+            nn.ReLU(inplace=True),
+            nn.Conv2d(feat_channel, out_channel, kernel_size=1))
+        return layer
+
+    def init_weights(self):
+        """Initialize weights of the head."""
+        bias_init = bias_init_with_prob(0.1)
```

Comment:
Should use init_cfg?

Reply:
init_cfg is difficult to implement due to `self.heatmap_head[-1]` only initialize bias.

Reply:
Add comments for that.